python/src/aac/context/language_context.py

```diff
@@ -115,6 +115,7 @@ def remove_definitions(self, definitions: list[Definition]) -> None:
         """Remove the given definitions from the context."""
         for definition in definitions:
             self.context_instance.definitions.remove(definition)
+            definition.source.is_loaded_in_context = False
```

Comment:
Similar here, this line actually needs to be first, so when the definition is updated within the context it has the updated flag.